app/uk/gov/hmrc/eusubsidycompliancefrontend/services/SubsidyJourney.scala

```diff
@@ -55,26 +53,29 @@ case class SubsidyJourney(
     cya
   )
 
-  val isAmend: Boolean = existingTransactionId.nonEmpty
+  val isAmend: Boolean = traderRef.value.nonEmpty
```

Comment:
Isn't trader ref an optional value? I think we initially chose existingTransactionId because it gets assigned on submission.

Reply:
It's optional but the value actually contains an object that tells us if a value was provided or not along with the value itself if provided. So, we are pretty much testing 'has the user answered the question on this page' here since the amend journey only needs to work for the create flow now. (This also fixes a bug where the amend flow on creating a payment did not work correctly).

Reply:
Oh cool. I assume `existingTransactionId` wasn't doing the trick then? Smashing. Approving.